### ISSUE TITLE
[a11y]: CreateFullPage component steps "span" and "button" target elements do not have minimum spacing requirements

### ISSUE THREAD

**Issue by anish-mm-05** [carbon-design-system/ibm-products#6288]
### Package

@carbon/ibm-products

### Browser

Chrome

### Operating System

MacOS

### Package version

v2.18.1

### React version

v18.2.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

CreateFullPage component steps has span and button target elements so it should meet a minimum spacing requirement of 12px.

![Image](https://github.com/user-attachments/assets/d51c4f82-c5c3-400c-a39f-e6d2a6e5e0c7)

### Reproduction/example

https://stackblitz.com/github/carbon-design-system/ibm-products/tree/main/examples/carbon-for-ibm-products/CreateFullPage?file=src%2FExample%2FExample.jsx

### Steps to reproduce

1. Open the example link or (https://wmaibzpqbagithub-qeho--5173--34c588ed.local-credentialless.webcontainer.io/)
2. Run the IBM Equal Access Accessibility Checker
3. Its shows an issue like this _Undersized target "span" does not have sufficient spacing of 12 CSS pixels from another target "button"_

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by ljcarot** [comment]
Working with accessibility team to confirm if this is a warning vs violation.

**Comment by amal-k-joy** [comment]
We observed the same issue in ProgressIndicator as well.
